From 80fd5acfd1c451337850adea9d5ef361f32e6c21 Mon Sep 17 00:00:00 2001
From: eppaneamd <rleppane@amd.com>
Date: Thu, 25 Sep 2025 12:54:05 +0000
Subject: [PATCH] Keep only one plus character in version suffix

---
 setup.py | 17 ++++++++++++++---
 1 file changed, 14 insertions(+), 3 deletions(-)

diff --git a/setup.py b/setup.py
index bf70d76ad..87d20cd62 100644
--- a/setup.py
+++ b/setup.py
@@ -774,9 +774,20 @@ def get_git_version_suffix():
     else:
         return get_git_commit_hash()
 
-
-# keep it separate for easy substitution
-TRITON_VERSION = "3.5.0" + get_git_version_suffix() + os.environ.get("TRITON_WHEEL_VERSION_SUFFIX", "")
+ 
+def get_triton_version_suffix():
+    # Either "" or "+<githash>", "<githash>" itself does not contain any plus-characters.
+    git_sfx = get_git_version_suffix()
+    # Should start with "+" that will replaced with "-" if needed
+    env_sfx = os.environ.get("TRITON_WHEEL_VERSION_SUFFIX", "")
+    # version suffix can only contain one plus-character
+    if "+" in git_sfx and "+" in env_sfx:
+        env_sfx = env_sfx.replace("+", "-")
+    return git_sfx + env_sfx
+
+
+ # keep it separate for easy substitution
+TRITON_VERSION = "3.5.0" + get_triton_version_suffix()
 
 # Dynamically define supported Python versions and classifiers
 MIN_PYTHON = (3, 9)
-- 
2.34.1

