diff --git a/aiter/ops/mha.py b/aiter/ops/mha.py
index 72be4ab21..43515c958 100644
--- a/aiter/ops/mha.py
+++ b/aiter/ops/mha.py
@@ -113,9 +113,10 @@ def common_mha_fwd_fake_tensors(
             head_size_v,
         ), "Output tensor has incorrect shape"
     else:
+        out_dtype = dtypes.bf16 if q.dtype == dtypes.fp8 else q.dtype
         out = torch.empty(
             (batch_size, seqlen_q, num_heads, head_size_v),
-            dtype=q.dtype,
+            dtype=out_dtype,
             device=q.device,
             requires_grad=q.requires_grad,
         )
