diff --git a/torchao/float8/inference.py b/torchao/float8/inference.py
index f15d38576..34fc10d12 100644
--- a/torchao/float8/inference.py
+++ b/torchao/float8/inference.py
@@ -19,6 +19,7 @@ from torchao.quantization.granularity import (
 )
 from torchao.utils import (
     is_MI300,
+    is_MI350,
     is_sm_at_least_89,
 )
 
@@ -109,9 +110,9 @@ def addmm_float8_unwrapped_inference(
             scale_a=a_scale,
             scale_b=b_scale,
             scale_result=output_scale,
-            out_dtype=output_dtype,
+            out_dtype=torch.bfloat16,
             use_fast_accum=use_fast_accum,
-        )
+        ).to(torch.float32)
         return output + bias
     return torch._scaled_mm(
         a_data,
@@ -120,9 +121,9 @@ def addmm_float8_unwrapped_inference(
         scale_b=b_scale,
         bias=bias,
         scale_result=output_scale,
-        out_dtype=output_dtype,
+        out_dtype=torch.bfloat16,
         use_fast_accum=use_fast_accum,
-    )
+    ).to(output_dtype)
 
 
 def _slice_scale_for_dimension(
@@ -249,6 +250,6 @@ def _check_hardware_support(
                 f"Invalid granularity type: {_granularity}, only PerTensor or PerRow are supported."
             )
 
-        assert is_sm_at_least_89() or is_MI300(), (
+        assert is_sm_at_least_89() or is_MI300() or is_MI350(), (
             "Float8 dynamic quantization requires CUDA compute capability ≥8.9 or MI300+."
         )
diff --git a/torchao/quantization/quant_api.py b/torchao/quantization/quant_api.py
index 67216a6d9..84b687d56 100644
--- a/torchao/quantization/quant_api.py
+++ b/torchao/quantization/quant_api.py
@@ -94,6 +94,7 @@ from torchao.quantization.weight_tensor_linear_activation_quantization import (
 from torchao.utils import (
     _is_fbgemm_genai_gpu_available,
     is_MI300,
+    is_MI350,
     is_sm_at_least_89,
     is_sm_at_least_90,
 )
@@ -1812,7 +1813,7 @@ def _float8_dynamic_activation_float8_weight_quantize_tensor(weight, config):
 def _float8_dynamic_activation_float8_weight_transform(
     module: torch.nn.Module, config: Float8DynamicActivationFloat8WeightConfig
 ):
-    assert is_sm_at_least_89() or is_MI300(), (
+    assert is_sm_at_least_89() or is_MI300() or is_MI350(), (
         "Float8 dynamic activation quantization is only supported on CUDA>=8.9 and MI300+"
     )
     if config.set_inductor_config:
@@ -1915,7 +1916,7 @@ float8_static_activation_float8_weight = Float8StaticActivationFloat8WeightConfi
 def _float8_static_activation_float8_weight_transform(
     module: torch.nn.Module, config: Float8StaticActivationFloat8WeightConfig
 ):
-    assert is_sm_at_least_89() or is_MI300(), (
+    assert is_sm_at_least_89() or is_MI300() or is_MI350(), (
         "Float8 static activation quantization is only supported on CUDA 8.9 and above"
     )
 
diff --git a/torchao/testing/training/roofline_utils.py b/torchao/testing/training/roofline_utils.py
index 6c51cef0b..666fa4054 100644
--- a/torchao/testing/training/roofline_utils.py
+++ b/torchao/testing/training/roofline_utils.py
@@ -54,6 +54,20 @@ gpu_name_to_specs = {
         # TODO(future): run measurement on hardware
         "pct_achievable_mem_bw": 0.92,
     },
+    # enabled for MI355x benchmarks
+    "AMD Radeon Graphics": {
+        # https://www.amd.com/en/products/accelerators/instinct/mi350/mi355x.html
+        "bf16_peak_tops": 2.5e15,
+        "fp8_peak_tops": 5.0e12,
+        # 5.3 TB per second
+        "peak_mem_bw_bytes_sec": 8.0e12,
+        # for now, copy over from H100
+        # TODO(future): run measurement on hardware
+        "pct_achievable_gemm_tops": 0.78,
+        # for now, copy over from H100
+        # TODO(future): run measurement on hardware
+        "pct_achievable_mem_bw": 0.92,
+    },
     "NVIDIA GeForce RTX 5090": {
         # https://images.nvidia.com/aem-dam/Solutions/geforce/blackwell/nvidia-rtx-blackwell-gpu-architecture.pdf
         "bf16_peak_tops": 209.5e12,
