diff --git a/yunchang/comm/all_to_all.py b/yunchang/comm/all_to_all.py
index 036d246..9f93366 100644
--- a/yunchang/comm/all_to_all.py
+++ b/yunchang/comm/all_to_all.py
@@ -35,70 +35,44 @@ def all_to_all_4D(
     seq_world_size = dist.get_world_size(group)
 
     if scatter_idx == 2 and gather_idx == 1:
-        # input (torch.tensor): a tensor sharded along dim 1 (bs, seqlen/P, hc, hs) output: (bs, seqlen, hc/P, hs)
         bs, shard_seqlen, hc, hs = input.shape
         seqlen = shard_seqlen * seq_world_size
         shard_hc = hc // seq_world_size
 
-        # transpose groups of heads with the seq-len parallel dimension, so that we can scatter them!
-        # (bs, seqlen/P, hc, hs) -reshape-> (bs, seq_len/P, P, hc/P, hs) -transpose(0,2)-> (P, seq_len/P, bs, hc/P, hs)
-        input_t = (
-            input.reshape(bs, shard_seqlen, seq_world_size, shard_hc, hs)
-            .transpose(0, 2)
-            .contiguous()
-        )
-
-        output = torch.empty_like(input_t)
-        # https://pytorch.org/docs/stable/distributed.html#torch.distributed.all_to_all_single
-        # (P, seq_len/P, bs, hc/P, hs) scatter seqlen -all2all-> (P, seq_len/P, bs, hc/P, hs) scatter head
+        input_view = input.view(bs, shard_seqlen, seq_world_size, shard_hc, hs)
+        input_permuted = input_view.permute(2, 1, 0, 3, 4).contiguous()
 
+        output = torch.empty_like(input_permuted)
+        
         if seq_world_size > 1:
-            dist.all_to_all_single(output, input_t, group=group)
+            dist.all_to_all_single(output, input_permuted, group=group)
             if use_sync:
                 torch.cuda.synchronize()
         else:
-            output = input_t
-        # if scattering the seq-dim, transpose the heads back to the original dimension
-        output = output.reshape(seqlen, bs, shard_hc, hs)
-
-        # (seq_len, bs, hc/P, hs) -reshape-> (bs, seq_len, hc/P, hs)
-        output = output.transpose(0, 1).contiguous().reshape(bs, seqlen, shard_hc, hs)
+            output = input_permuted
 
+        output = output.permute(2, 0, 1, 3, 4).contiguous().view(bs, seqlen, shard_hc, hs)
+        
         return output
-
     elif scatter_idx == 1 and gather_idx == 2:
-        # input (torch.tensor): a tensor sharded along dim 1 (bs, seqlen, hc/P, hs) output: (bs, seqlen/P, hc, hs)
         bs, seqlen, shard_hc, hs = input.shape
         hc = shard_hc * seq_world_size
         shard_seqlen = seqlen // seq_world_size
-        seq_world_size = dist.get_world_size(group)
+        
+        input_view = input.view(bs, seq_world_size, shard_seqlen, shard_hc, hs)
+        input_permuted = input_view.permute(1, 3, 2, 0, 4).contiguous()
 
-        # transpose groups of heads with the seq-len parallel dimension, so that we can scatter them!
-        # (bs, seqlen, hc/P, hs) -reshape-> (bs, P, seq_len/P, hc/P, hs) -transpose(0, 3)-> (hc/P, P, seqlen/P, bs, hs) -transpose(0, 1) -> (P, hc/P, seqlen/P, bs, hs)
-        input_t = (
-            input.reshape(bs, seq_world_size, shard_seqlen, shard_hc, hs)
-            .transpose(0, 3)
-            .transpose(0, 1)
-            .contiguous()
-            .reshape(seq_world_size, shard_hc, shard_seqlen, bs, hs)
-        )
+        output = torch.empty_like(input_permuted)
 
-        output = torch.empty_like(input_t)
-        # https://pytorch.org/docs/stable/distributed.html#torch.distributed.all_to_all_single
-        # (P, bs x hc/P, seqlen/P, hs) scatter seqlen -all2all-> (P, bs x seq_len/P, hc/P, hs) scatter head
         if seq_world_size > 1:
-            dist.all_to_all_single(output, input_t, group=group)
+            dist.all_to_all_single(output, input_permuted, group=group)
             if use_sync:
                 torch.cuda.synchronize()
         else:
-            output = input_t
-
-        # if scattering the seq-dim, transpose the heads back to the original dimension
-        output = output.reshape(hc, shard_seqlen, bs, hs)
-
-        # (hc, seqlen/N, bs, hs) -tranpose(0,2)-> (bs, seqlen/N, hc, hs)
-        output = output.transpose(0, 2).contiguous().reshape(bs, shard_seqlen, hc, hs)
+            output = input_permuted
 
+        output = output.permute(3, 2, 0, 1, 4).contiguous().view(bs, shard_seqlen, hc, hs)
+        
         return output
     else:
         raise RuntimeError("scatter_idx must be 1 or 2 and gather_idx must be 1 or 2")
diff --git a/yunchang/globals.py b/yunchang/globals.py
index 6c650ab..7ff6de4 100644
--- a/yunchang/globals.py
+++ b/yunchang/globals.py
@@ -1,3 +1,4 @@
+
 import torch
 import os
 
@@ -109,6 +110,13 @@ try:
 except ImportError:
     HAS_FLASHINFER = False
 
+try:
+    import aiter
+    from aiter import flash_attn_func as flash_attn_func_aiter
+    HAS_AITER = True
+except ImportError:
+    HAS_AITER = False
+
 try:
     import sageattention
     HAS_SAGE_ATTENTION = True
@@ -120,4 +128,3 @@ try:
     HAS_SPARSE_SAGE_ATTENTION = True
 except ImportError:
     HAS_SPARSE_SAGE_ATTENTION = False
-
diff --git a/yunchang/kernels/__init__.py b/yunchang/kernels/__init__.py
index f2ac2a8..2157785 100644
--- a/yunchang/kernels/__init__.py
+++ b/yunchang/kernels/__init__.py
@@ -1,7 +1,9 @@
+
 from functools import partial
 
 import torch
 from .attention import (
+    flash_attn_forward_aiter,
     flash_attn_forward,
     flash_attn_backward,
     flash_attn3_func_forward,
@@ -15,6 +17,7 @@ from .attention import (
 from enum import Enum, auto
 
 from yunchang.globals import (
+    HAS_AITER,
     HAS_FLASH_ATTN,
     HAS_SAGE_ATTENTION,
     HAS_SPARSE_SAGE_ATTENTION,
@@ -31,6 +34,7 @@ if HAS_SPARSE_SAGE_ATTENTION:
 
 
 class AttnType(Enum):
+    AITER = "aiter"
     FA = "fa"
     FA3 = "fa3"
     FLASHINFER = "flashinfer"
@@ -53,7 +57,17 @@ class AttnType(Enum):
 def select_flash_attn_impl(
     impl_type: AttnType, stage: str = "fwd-bwd", attn_processor: torch.nn.Module = None
 ):
-    if impl_type == AttnType.FA:
+    if impl_type == AttnType.AITER:
+        if stage == "fwd-only":
+            return flash_attn_forward_aiter
+        elif stage == "bwd-only":
+            raise ValueError("Aiter does not support bwd-only stage.")
+        elif stage == "fwd-bwd":
+            raise ValueError("Aiter does not support fwd-bwd stage.")
+        else:
+            raise ValueError(f"Unknown stage: {stage}")
+
+    elif impl_type == AttnType.FA:
         if stage == "fwd-only":
             return flash_attn_forward
         elif stage == "bwd-only":
diff --git a/yunchang/kernels/attention.py b/yunchang/kernels/attention.py
index 02c6fce..7e48e8e 100644
--- a/yunchang/kernels/attention.py
+++ b/yunchang/kernels/attention.py
@@ -15,7 +15,11 @@ try:
 except ModuleNotFoundError:
     pass
 
-from yunchang.globals import HAS_FLASH_ATTN, HAS_FLASH_ATTN_HOPPER, HAS_FLASHINFER
+from yunchang.globals import HAS_FLASH_ATTN, HAS_FLASH_ATTN_HOPPER, HAS_FLASHINFER, HAS_AITER
+
+if HAS_AITER:
+    import aiter
+    from aiter import flash_attn_func as flash_attn_func_aiter
 
 if HAS_FLASH_ATTN:
     import flash_attn
@@ -79,7 +83,7 @@ def pytorch_attn_forward(
         )[:2]
     else:
         raise ValueError(f"Invalid op_type: {op_type}")
-    
+
     out = out.transpose(1, 2)
     lse = lse.to(q.dtype)
     return out, lse
@@ -111,47 +115,63 @@ def pytorch_attn_backward(
     # https://github.com/pytorch/pytorch/blob/main/tools/autograd/derivatives.yaml#L2874
 
 
-def flash_attn_forward(q, k, v, 
-        dropout_p = 0.0, 
-        softmax_scale = None, 
-        causal=False, 
-        window_size=(-1, -1), 
-        softcap=None, 
-        alibi_slopes=None, 
+def flash_attn_forward(q, k, v,
+        dropout_p = 0.0,
+        softmax_scale = None,
+        causal=False,
+        window_size=(-1, -1),
+        softcap=None,
+        alibi_slopes=None,
         return_softmax=False):
-    assert HAS_FLASH_ATTN, "FlashAttention is not available"
     if softmax_scale is None:
         softmax_scale = q.shape[-1] ** (-0.5)
-    if flash_attn.__version__ < '2.6.3':
-        block_out, _, _, _, _, block_lse, _, _ = _flash_attn_forward(
+    if HAS_AITER:
+        block_out, block_lse, _, _ = aiter.ops.mha._flash_attn_forward(
             q,
             k,
             v,
-            dropout_p = dropout_p,
-            softmax_scale = softmax_scale,
-            causal=causal,
-            window_size=window_size,
-            softcap=softcap,
-            alibi_slopes=alibi_slopes,
-            return_softmax=return_softmax,
-        )
-    else:
-        block_out, block_lse, _, _ = _flash_attn_forward(
-            q,
-            k,
-            v,
-            dropout_p = dropout_p,
-            softmax_scale = softmax_scale,
+            dropout_p=dropout_p,
+            softmax_scale=softmax_scale,
             causal=causal,
             window_size_left=window_size[0],
             window_size_right=window_size[1],
-            softcap=softcap,
+            bias=None,
             alibi_slopes=alibi_slopes,
+            return_lse=True,
             return_softmax=return_softmax,
         )
+    else:
+        assert HAS_FLASH_ATTN, "FlashAttention is not available"
+        if flash_attn.__version__ < '2.6.3':
+            block_out, _, _, _, _, block_lse, _, _ = _flash_attn_forward(
+                q,
+                k,
+                v,
+                dropout_p = dropout_p,
+                softmax_scale = softmax_scale,
+                causal=causal,
+                window_size=window_size,
+                softcap=softcap,
+                alibi_slopes=alibi_slopes,
+                return_softmax=return_softmax,
+            )
+        else:
+            block_out, block_lse, _, _ = _flash_attn_forward(
+                q,
+                k,
+                v,
+                dropout_p = dropout_p,
+                softmax_scale = softmax_scale,
+                causal=causal,
+                window_size_left=window_size[0],
+                window_size_right=window_size[1],
+                softcap=softcap,
+                alibi_slopes=alibi_slopes,
+                return_softmax=return_softmax,
+            )
     return block_out, block_lse
 
-def flash_attn_backward(dout, q, k, v, out, softmax_lse, block_dq_buffer, block_dk_buffer, block_dv_buffer, dropout_p, softmax_scale, 
+def flash_attn_backward(dout, q, k, v, out, softmax_lse, block_dq_buffer, block_dk_buffer, block_dv_buffer, dropout_p, softmax_scale,
     bwd_causal, window_size, softcap, alibi_slopes, deterministic, rng_state):
     if softmax_scale is None:
         softmax_scale = q.shape[-1] ** (-0.5)
@@ -197,7 +217,7 @@ def flash_attn_backward(dout, q, k, v, out, softmax_lse, block_dq_buffer, block_
             deterministic,
             rng_state,
         )
-    
+
 
 def flash_attn3_func_forward(q, k, v, dropout_p, softmax_scale, causal, window_size, softcap, alibi_slopes, return_softmax):
     assert HAS_FLASH_ATTN_HOPPER
@@ -239,12 +259,12 @@ def flash_attn3_func_forward(q, k, v, dropout_p, softmax_scale, causal, window_s
                     pack_gqa=None,
                     sm_margin=0,
                 )
-    
+
     return out, softmax_lse
 
-def flash_attn3_func_backward(dout, q, k, v, out, softmax_lse, 
-                                    block_dq_buffer, block_dk_buffer, block_dv_buffer, 
-                                    dropout_p, softmax_scale, 
+def flash_attn3_func_backward(dout, q, k, v, out, softmax_lse,
+                                    block_dq_buffer, block_dk_buffer, block_dv_buffer,
+                                    dropout_p, softmax_scale,
                                     bwd_causal, window_size, softcap, alibi_slopes, deterministic, rng_state):
     # (dout, q, k, v, out, softmax_lse, dq, dk, dv, softmax_scale, causal):
     assert HAS_FLASH_ATTN_HOPPER, f"FlashAttention Hopper is not available"
@@ -273,6 +293,30 @@ def flash_attn3_func_backward(dout, q, k, v, out, softmax_lse,
         sm_margin=0,
     )
 
+def flash_attn_forward_aiter(q, k, v,
+    dropout_p = 0.0,
+    softmax_scale = None,
+    causal=False,
+    window_size=(-1, -1),
+    softcap=None,
+    alibi_slopes=None,
+    return_softmax=False
+):
+    assert HAS_AITER, "Aiter is not available"
+    block_out, block_lse = flash_attn_func_aiter(
+        q,
+        k,
+        v,
+        dropout_p = dropout_p,
+        softmax_scale = softmax_scale,
+        causal = causal,
+        window_size=window_size,
+        alibi_slopes = alibi_slopes,
+        return_lse=True,
+    )
+
+    return block_out, block_lse
+
 def flashinfer_attn_forward(
     q: torch.Tensor,
     k: torch.Tensor,
