diff --git a/examples/hunyuan_video_usp_example.py b/examples/hunyuan_video_usp_example.py
index b1d6fce..7470839 100644
--- a/examples/hunyuan_video_usp_example.py
+++ b/examples/hunyuan_video_usp_example.py
@@ -92,14 +92,8 @@ def parallelize_transformer(pipe: DiffusionPipeline):
                                     get_sequence_parallel_world_size(),
                                     dim=-2)[get_sequence_parallel_rank()]
 
-        encoder_attention_mask = encoder_attention_mask[0].to(torch.bool)
-        encoder_hidden_states_indices = torch.arange(
-            encoder_hidden_states.shape[1],
-            device=encoder_hidden_states.device)
-        encoder_hidden_states_indices = encoder_hidden_states_indices[
-            encoder_attention_mask]
-        encoder_hidden_states = encoder_hidden_states[
-            ..., encoder_hidden_states_indices, :]
+        encoder_attention_mask = encoder_attention_mask.to(torch.bool).any(dim=0)
+        encoder_hidden_states = encoder_hidden_states[:, encoder_attention_mask, :]
         if encoder_hidden_states.shape[-2] % get_sequence_parallel_world_size(
         ) != 0:
             get_runtime_state().split_text_embed_in_sp = False
@@ -234,7 +228,7 @@ def main():
         height=input_config.height,
         width=input_config.width,
         num_frames=input_config.num_frames,
-        batch_size=1,
+        batch_size=input_config.batch_size,
         num_inference_steps=input_config.num_inference_steps,
         split_text_embed_in_sp=get_pipeline_parallel_world_size() == 1,
     )
@@ -270,8 +264,7 @@ def main():
 
     if engine_config.runtime_config.use_torch_compile:
         torch._inductor.config.reorder_for_compute_comm_overlap = True
-        pipe.transformer = torch.compile(pipe.transformer,
-                                         mode="max-autotune-no-cudagraphs")
+        pipe.transformer.compile()
 
         # one step to warmup the torch compiler
         output = pipe(
@@ -282,7 +275,7 @@ def main():
             num_inference_steps=1,
             generator=torch.Generator(device="cuda").manual_seed(
                 input_config.seed),
-        ).frames[0]
+        )
 
     torch.cuda.reset_peak_memory_stats()
     start_time = time.time()
@@ -295,7 +288,7 @@ def main():
         num_inference_steps=input_config.num_inference_steps,
         generator=torch.Generator(device="cuda").manual_seed(
             input_config.seed),
-    ).frames[0]
+    )
 
     end_time = time.time()
     elapsed_time = end_time - start_time
@@ -309,9 +302,10 @@ def main():
     )
     if is_dp_last_group():
         resolution = f"{input_config.width}x{input_config.height}"
-        output_filename = f"results/hunyuan_video_{parallel_info}_{resolution}.mp4"
-        export_to_video(output, output_filename, fps=15)
-        print(f"output saved to {output_filename}")
+        for idx, frames in enumerate(output.frames, start=1):
+            output_filename = f"results/hunyuan_video_{idx:02d}_{parallel_info}_{resolution}.mp4"
+            export_to_video(frames, output_filename, fps=15)
+            print(f"output saved to {output_filename}")
 
     if get_world_group().rank == get_world_group().world_size - 1:
         print(
diff --git a/xfuser/envs.py b/xfuser/envs.py
index 77a947e..e782e7b 100644
--- a/xfuser/envs.py
+++ b/xfuser/envs.py
@@ -78,11 +78,28 @@ class PackagesEnvChecker:
 
     def initialize(self):
         self.packages_info = {
+            "has_aiter": self.check_aiter(),
             "has_flash_attn": self.check_flash_attn(),
             "has_long_ctx_attn": self.check_long_ctx_attn(),
             "diffusers_version": self.check_diffusers_version(),
         }
 
+    def check_aiter(self):
+        """
+        Checks whether ROCm AITER library is installed
+        """
+        try:
+            import aiter
+            logger.info("Using AITER as the attention library")
+            return True
+        except:
+            if _is_hip():
+                logger.warning(
+                    f'Using AMD GPUs, but library "aiter" is not installed, '
+                    'defaulting to other attention mechanisms'
+                )
+            return False
+
     def check_flash_attn(self):
         try:
             device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
@@ -94,12 +111,12 @@ class PackagesEnvChecker:
                 from flash_attn import __version__
 
                 if __version__ < "2.6.0":
-                    raise ImportError(f"install flash_attn >= 2.6.0")
+                    raise ImportError(f'install flash_attn >= 2.6.0')
                 return True
         except ImportError:
             logger.warning(
                 f'Flash Attention library "flash_attn" not found, '
-                f"using pytorch attention implementation"
+                f'defaulting to other attention mechanisms'
             )
             return False
 
diff --git a/xfuser/model_executor/layers/attention_processor.py b/xfuser/model_executor/layers/attention_processor.py
index 77f0bb1..a2022c6 100644
--- a/xfuser/model_executor/layers/attention_processor.py
+++ b/xfuser/model_executor/layers/attention_processor.py
@@ -49,6 +49,7 @@ logger = init_logger(__name__)
 
 env_info = PACKAGES_CHECKER.get_packages_info()
 HAS_LONG_CTX_ATTN = env_info["has_long_ctx_attn"]
+HAS_AITER = env_info["has_aiter"]
 HAS_FLASH_ATTN = env_info["has_flash_attn"]
 
 
@@ -1357,14 +1358,25 @@ if HunyuanVideoAttnProcessor2_0 is not None:
                 from xfuser.core.long_ctx_attention import (
                     xFuserLongContextAttention,
                 )
+                if HAS_AITER:
+                    import yunchang
+                    from yunchang.kernels import AttnType
 
-                if HAS_FLASH_ATTN:
+                    assert yunchang.__version__ >= "0.6.3"
+                    self.hybrid_seq_parallel_attn = xFuserLongContextAttention(
+                        use_kv_cache=self.use_long_ctx_attn_kvcache,
+                        attn_type=AttnType.AITER,
+                    )
+                elif HAS_FLASH_ATTN:
+                    # self.hybrid_seq_parallel_attn = LongContextAttention()
                     self.hybrid_seq_parallel_attn = xFuserLongContextAttention(
                         use_kv_cache=self.use_long_ctx_attn_kvcache
                     )
                 else:
+                    import yunchang
                     from yunchang.kernels import AttnType
 
+                    assert yunchang.__version__ >= "0.6.0"
                     self.hybrid_seq_parallel_attn = xFuserLongContextAttention(
                         use_kv_cache=self.use_long_ctx_attn_kvcache,
                         attn_type=AttnType.TORCH,
diff --git a/xfuser/model_executor/layers/usp.py b/xfuser/model_executor/layers/usp.py
index b4ca713..fc54afa 100644
--- a/xfuser/model_executor/layers/usp.py
+++ b/xfuser/model_executor/layers/usp.py
@@ -1,6 +1,7 @@
 # This file implements USP with torch version >= '2.5.0'
 import torch
 from torch.nn import functional as F
+from torch.nn.attention import sdpa_kernel
 
 import torch.distributed._functional_collectives as ft_c
 
@@ -14,26 +15,43 @@ from xfuser.core.distributed import (
     get_ring_parallel_world_size,
 )
 
+from packaging.version import parse
 from xfuser.envs import PACKAGES_CHECKER
 env_info = PACKAGES_CHECKER.get_packages_info()
 HAS_FLASH_ATTN = env_info["has_flash_attn"]
+if HAS_FLASH_ATTN:
+    import flash_attn
+
+HAS_AITER = env_info["has_aiter"]
+if HAS_AITER:
+    import aiter
 
 aten = torch.ops.aten
 
     
 def ring_attn(query, key, value, dropout_p=0.0, is_causal=False):
-    if torch.__version__ >= "2.6.0":
+    if parse(torch.__version__).release >= parse("2.6.0").release:
         from torch.distributed.tensor.experimental._attention import _cp_options
         _cp_options.enable_load_balance = False
         kwargs = {
             "dropout_p": dropout_p,
             "is_causal": is_causal,
         }
-        if HAS_FLASH_ATTN:
+        if HAS_AITER:
             out, *_ = _templated_ring_attention(
                 PROCESS_GROUP.RING_PG,
                 1,
-                aten._scaled_dot_product_flash_attention,
+                _aiter_attn_call,
+                query,
+                key,
+                value,
+                **kwargs,
+            )
+        elif HAS_FLASH_ATTN:
+            out, *_ = _templated_ring_attention(
+                PROCESS_GROUP.RING_PG,
+                1,
+                _flash_attn_call,
                 query,
                 key,
                 value,
@@ -48,7 +66,7 @@ def ring_attn(query, key, value, dropout_p=0.0, is_causal=False):
             out, *_ = _templated_ring_attention(
                 PROCESS_GROUP.RING_PG,
                 1,
-                aten._scaled_dot_product_efficient_attention,
+                aten._scaled_dot_product_flash_attention,
                 query,
                 key,
                 value,
@@ -59,10 +77,20 @@ def ring_attn(query, key, value, dropout_p=0.0, is_causal=False):
             "dropout_p": dropout_p,
             "is_causal": is_causal,
         }
-        if HAS_FLASH_ATTN:
+        if HAS_AITER:
             out, *_ = _templated_ring_attention(
                 PROCESS_GROUP.RING_PG,
-                aten._scaled_dot_product_flash_attention,
+                1,
+                _aiter_attn_call,
+                query,
+                key,
+                value,
+                **kwargs,
+            )
+        elif HAS_FLASH_ATTN:
+            out, *_ = _templated_ring_attention(
+                PROCESS_GROUP.RING_PG,
+                _flash_attn_call,
                 query,
                 key,
                 value,
@@ -76,7 +104,7 @@ def ring_attn(query, key, value, dropout_p=0.0, is_causal=False):
             }
             out, *_ = _templated_ring_attention(
                 PROCESS_GROUP.RING_PG,
-                aten._scaled_dot_product_efficient_attention,
+                aten._scaled_dot_product_flash_attention,
                 query,
                 key,
                 value,
@@ -133,12 +161,63 @@ def _ft_c_output_all_to_all(x):
     x = x.reshape(world_size, s // world_size, b, -1, d).permute(2, 0, 3, 1, 4).reshape(b, -1, s // world_size, d)
     return x
 
+def _aiter_attn_call(query, key, value, dropout_p, is_causal):
+    """
+    Performs the necessary tensor permutes and
+    then calls attention through AITER
+    """
+    query = torch.permute(query, [0, 2, 1, 3]).contiguous()
+    key = torch.permute(key, [0, 2, 1, 3]).contiguous()
+    value = torch.permute(value, [0, 2, 1, 3]).contiguous()
+    output, softmax_lse = aiter.flash_attn_func(
+        query,
+        key,
+        value,
+        dropout_p=dropout_p,
+        causal=is_causal,
+        return_attn_probs=False,
+        return_lse=True
+    )
+    output = torch.permute(output, [0, 2, 1, 3])
+    return output, softmax_lse
+
+def _flash_attn_call(query, key, value, dropout_p, is_causal):
+    """
+    Performs the necessary tensor permutes and
+    then calls attention through flash_attn
+    """
+    query = torch.permute(query, [0, 2, 1, 3])
+    key = torch.permute(key, [0, 2, 1, 3])
+    value = torch.permute(value, [0, 2, 1, 3])
+    output, softmax_lse, S_mask = flash_attn.flash_attn_func(
+        query,
+        key,
+        value,
+        dropout_p=dropout_p,
+        causal=is_causal,
+        return_attn_probs=True,
+    )
+    output = torch.permute(output, [0, 2, 1, 3])
+    return output, softmax_lse
+
+def _attention(query, key, value, dropout_p, is_causal):
+    """
+    Calls the correct attention mechanism based on the available libraries
+    """
+    if HAS_AITER:
+        output, _ = _aiter_attn_call(query, key, value, dropout_p, is_causal)
+        return output
+    elif HAS_FLASH_ATTN:
+        output, _ = _flash_attn_call(query, key, value, dropout_p, is_causal)
+        return output
+    else:
+        with sdpa_kernel(torch.nn.attention.SDPBackend.FLASH_ATTENTION):
+            output = F.scaled_dot_product_attention(query, key, value, dropout_p=dropout_p, is_causal=is_causal)
+        return output
 
 def USP(query, key, value, dropout_p=0.0, is_causal=False):
     if get_sequence_parallel_world_size() == 1:
-        out = F.scaled_dot_product_attention(
-            query, key, value, dropout_p=dropout_p, is_causal=is_causal
-        )
+        out = _attention(query, key, value, dropout_p=dropout_p, is_causal=is_causal)
     elif get_ulysses_parallel_world_size() == 1:
         out = ring_attn(query, key, value, dropout_p=dropout_p, is_causal=is_causal)
     elif get_ulysses_parallel_world_size() > 1:
@@ -147,9 +226,7 @@ def USP(query, key, value, dropout_p=0.0, is_causal=False):
         value = _ft_c_input_all_to_all(value)
 
         if get_ring_parallel_world_size() == 1:
-            out = F.scaled_dot_product_attention(
-                query, key, value, dropout_p=dropout_p, is_causal=is_causal
-            )
+            out = _attention(query, key, value, dropout_p=dropout_p, is_causal=is_causal)
         else:
             out = ring_attn(query, key, value, dropout_p=dropout_p, is_causal=is_causal)
 
