diff --git a/xfuser/core/distributed/attention_backend.py b/xfuser/core/distributed/attention_backend.py
index b23cda9..9875552 100644
--- a/xfuser/core/distributed/attention_backend.py
+++ b/xfuser/core/distributed/attention_backend.py
@@ -1,3 +1,4 @@
+import functools
 import torch
 import torch.nn.functional as F
 from enum import Enum
@@ -8,6 +9,7 @@ ATTENTION_FUNCTION_REGISTRY = {}
 aten = torch.ops.aten
 env_info = PACKAGES_CHECKER.get_packages_info()
 if env_info["has_aiter"]:
+    import aiter
     from aiter import flash_attn_func as flash_attn_func_aiter
     import inspect
     try:
@@ -33,8 +35,10 @@ class AttentionBackendType(Enum):
     FLASH = "Flash Attention V2"
     CUDNN =  "cuDNN"
     FLASH_3 = "Flash Attention V3"
+    FLASH_3_FP8 = "Flash Attention v3 FP8"
     FLASH_4 = "Flash Attention V4"
     AITER = "AITER"
+    AITER_FP8 = "AITER FP8"
 
 def register_attention_function(backend_type):
     """
@@ -137,6 +141,52 @@ def _flash_attn_3_call(query, key, value, dropout_p, is_causal):
     output = torch.permute(output, [0, 2, 1, 3])
     return output, softmax_lse
 
+@functools.lru_cache()
+def get_dtype_max(dtype):
+    try:
+        dtypeMax = torch.finfo(dtype).max
+    except:
+        dtypeMax = torch.iinfo(dtype).max
+    return dtypeMax
+
+def per_tensor_quant(
+    x, scale=None, scale_dtype=torch.float32, quant_dtype=torch.float8_e4m3fn, dtypeMax=None
+):
+    x = x.to(torch.float32)
+    if scale is None:
+        if dtypeMax is None:
+            dtypeMax = get_dtype_max(quant_dtype)
+        scale = torch.abs(x).max() / dtypeMax
+    y = x / scale
+    return y.to(quant_dtype), scale.expand(*x.shape[:2]).to(scale_dtype)
+
+@register_attention_function(AttentionBackendType.FLASH_3_FP8)
+def _flash_attn_3_fp8_call(query, key, value, dropout_p, is_causal):
+    """
+    Performs the necessary tensor permutes and
+    then calls attention through flash_attn V3
+    """
+    # quantize
+    query, scale_query = per_tensor_quant(query)
+    key, scale_key = per_tensor_quant(key)
+    value, scale_value = per_tensor_quant(value)
+    # run
+    query = torch.permute(query, [0, 2, 1, 3]).contiguous()
+    key = torch.permute(key, [0, 2, 1, 3]).contiguous()
+    value = torch.permute(value, [0, 2, 1, 3]).contiguous()
+    output, softmax_lse = flash_attn_func_3(
+        query,
+        key,
+        value,
+        causal=is_causal,
+        return_attn_probs=True,
+        q_descale=scale_query,
+        k_descale=scale_key,
+        v_descale=scale_value,
+    )
+    output = torch.permute(output, [0, 2, 1, 3])
+    return output, softmax_lse
+
 @register_attention_function(AttentionBackendType.FLASH_4)
 @torch.compiler.disable # Disabling compile, as it is not currently supported with FAv4
 def _flash_attn_4_call(query, key, value, dropout_p, is_causal):
@@ -157,6 +207,30 @@ def _flash_attn_4_call(query, key, value, dropout_p, is_causal):
     output = torch.permute(output, [0, 2, 1, 3])
     return output, softmax_lse
 
+@register_attention_function(AttentionBackendType.AITER_FP8)
+def _aiter_fp8_attn_call(query, key, value, dropout_p, is_causal):
+    """
+    Performs the necessary tensor permutes and
+    then calls attention through AITER
+    """
+    query = torch.permute(query, [0, 2, 1, 3]).contiguous()
+    key = torch.permute(key, [0, 2, 1, 3]).contiguous()
+    value = torch.permute(value, [0, 2, 1, 3]).contiguous()
+
+    softmax_lse = None
+    quant_dtype = aiter.dtypes.fp8
+    # Descale not yet supported in AITER.
+    quant_q, _ = aiter.per_tensor_quant(query, scale=torch.tensor(1, device=query.device), quant_dtype=quant_dtype)
+    quant_k, _ = aiter.per_tensor_quant(key, scale=torch.tensor(1, device=key.device), quant_dtype=quant_dtype)
+    quant_v, _ = aiter.per_tensor_quant(value, scale=torch.tensor(1, device=value.device), quant_dtype=quant_dtype)
+    torch._dynamo.graph_break() # Without this line, fp8 call will crash due to q and k dont have the same dtype, which is not true.
+    output = aiter.flash_attn_fp8_pertensor_func(
+        quant_q, quant_k, quant_v,
+        causal=is_causal,
+    )
+    output = torch.permute(output, [0, 2, 1, 3])
+    return output, softmax_lse
+
 @register_attention_function(AttentionBackendType.AITER)
 def _aiter_attn_call(query, key, value, dropout_p, is_causal):
     """
