diff --git a/xfuser/core/distributed/attention_backend.py b/xfuser/core/distributed/attention_backend.py
index 2547b99..fbab93a 100644
--- a/xfuser/core/distributed/attention_backend.py
+++ b/xfuser/core/distributed/attention_backend.py
@@ -251,7 +251,6 @@ def _aiter_fp8_attn_call(query, key, value, dropout_p, is_causal):
                 "k_descale": k_descale,
                 "v_descale": v_descale,
             }
-    torch._dynamo.graph_break()
     output = aiter.flash_attn_fp8_pertensor_func(
         quant_q, quant_k, quant_v,
         causal=is_causal,
